apps/src/netsim/NetSimLobby.js

```diff
@@ -0,0 +1,240 @@
+
+/**
+ * @param {NetSimConnection} connection - The instance connection that this
+ *                           lobby control will manipulate.
+ * @constructor
+ */
+var NetSimLobby = function (connection) {
+
+  /**
+   * Instance connection that this lobby control will manipulate.
+   * @type {NetSimConnection}
+   * @private
+   */
+  this.connection_ = connection;
+  this.connection_.statusChanges.register(this, this.refreshLobby_);
+
+  /**
+   * When the lobby should be refreshed next
+   * @type {Number}
+   * @private
+   */
+  this.nextAutoRefreshTime_ = Infinity;
+};
+module.exports = NetSimLobby;
+
+/**
+ * Generate a new NetSimLobby object, putting
+ * its markup within the provided element and returning
+ * the controller object.
+ * @param {DOMElement} The container for the lobby markup
+ * @param {NetSimConnection} The connection manager to use
+ * @return {NetSimLobby} A new controller for the generated lobby
+ * @static
+ */
+NetSimLobby.createWithin = function (element, connection) {
+  // Create a new NetSimLobby
+  var controller = new NetSimLobby(connection);
+  // TODO: Figure out what parameters to pass here
+  element.innerHTML = markup({});
+  controller.initialize();
+  return controller;
+};
+
+/**
+ *
+ */
+NetSimLobby.prototype.initialize = function () {
+  this.bindElements_();
+  this.refreshInstanceList_();
+};
+
+/**
+ * Grab the DOM elements related to this control -once-
+ * and bind them to member variables.
+ * Also attach method handlers.
+ */
+NetSimLobby.prototype.bindElements_ = function () {
+  this.instanceSelector_ = document.getElementById('netsim_instance_select');
+  this.instanceSelector_.addEventListener('change',
+      _.bind(this.onInstanceSelectorChange_, this));
+
+  this.lobbyList_ = document.getElementById('netsim_lobby_list');
+};
+
+/**
+ *
+ */
+NetSimLobby.prototype.onInstanceSelectorChange_ = function () {
+  if (this.connection_.isConnectedToInstance()) {
+    this.connection_.disconnectFromInstance();
+    this.nextAutoRefreshTime_ = Infinity;
+  }
+
+  if (this.instanceSelector_.value !== '__none') {
+    this.connection_.connectToInstance(this.instanceSelector_.value);
+  }
+};
+
+/**
+ * Make an async request against the dashboard API to
+ * reload and populate the user sections list.
+ */
+NetSimLobby.prototype.refreshInstanceList_ = function () {
+  var option;
+  var self = this;
+  var instanceSelector = this.instanceSelector_;
+  this.getUserSections_(function (data) {
+    $(instanceSelector).empty();
+
+    if (0 === data.length){
+      // If we didn't get any sections, we must deny access
+      option = document.createElement('option');
+      option.value = '__none';
+      option.textContent = '-- NONE FOUND --';
+      instanceSelector.appendChild(option);
+      return;
+    } else {// if (data.length > 1) {
+      // If we have more than one section, require the user
+      // to pick one.
+      option = document.createElement('option');
```

Comment:
Oops, should have commented this.  Eventually, students with only one section will auto-connect to that section on load and be unable to disconnect.  The commented `else if` here ensures that we always have a "-- PICK ONE --" option so it's easy to test connecting and disconnecting, even if your student account has only one section.